es-ds-components/components/es-dropdown-select.vue

```diff
@@ -33,7 +33,10 @@ const isInputClicked = ref(false);
 const isLabelClicked = ref(false);
 const isDisabled = computed(() => props.disabled || props.options.length === 0);
 
-const isSelected = (option: any) => option === props.modelValue;
+const isSelected = (option: any) => {
```

Comment:
This is an elegant solution, but just to throw it out there, there's also a vue3 function https://vuejs.org/api/reactivity-utilities.html#unref that could potentially be helpful in situations like this.